### DISCUSSION TITLE
Need to understand more about capacitor

### DISCUSSION THREAD

**Discussion by Samnan** [ionic-team/capacitor#3659] [Q&A] [answered]
Hello everyone,

   I have read through capacitor docs with great interest in the past few days. Also played around by converting two of my past apps to capacitor (not completely tested on device/emulator, but web based testing looks fine).

   My questions, if someone with good knowledge of capacitor can answer, would be grateful.

* What exactly is all the fuss about capacitor being more "native" than cordova. As far as I see, all the application code still is web based and if we use cordova camera plugin, the camera is invoked through native device camera. So I would like to know the technical details of how this "native" experience of capacitor differs from cordova app.
* OneSignal is probably most favourite push notification plugin, and PN are needed in almost 99% of apps. So why cap does not support ~~Tony~~ OneSignal out of the box like other cordova plugins.
* If cap is supposed to bring new goodness to hybrid development, why does it try to be compatible with cordova and support almost 90% of plugins and api, while deprecating few good ones (OneSignal). Why not have its own streamlined apis for all plugins and be a total alternative for cordova. We have RN/Flutter/Xamarin as an example. People who converted their native apps to RN are happy people (mostly), and if cap has that good stuff in it, i would surely use it for my next app and even rewrite existing cordova code. Anyway its mostly the native functionality in my web app, which is like 20% of the app code.

   Thanks.

**Comment by mlynch** [comment]
For your first question, not sure where you've read that. Capacitor handles your native projects differently than Cordova so it's much easier to add native functionality when needed, while Cordova abstracts away a lot of your native projects making it harder to do one-off things in them.

For one signal, see https://documentation.onesignal.com/docs/ionic-sdk-setup

Capacitor supports Cordova plugins because many of them work really well and they are native code anyways, so supporting them isn't a negative for Capacitor in any way. Capacitor _does_ have it streamlined API for building plugins if you only care about supporting Capacitor.

**Comment by Samnan** [comment]
@mlynch thanks so much for finally providing some much needed info. Two quick things:
- Is it possible to provide a link or more information about the first point with some more technical details? Maybe someone did a blog post?
- I would still suggest to ignore cordova support completely and let capacitor be its own master, just my two cents :)

**Comment by mlynch** [comment, marked as answer]
More info on Capacitor vs Cordova: https://capacitorjs.com/docs/cordova

Appreciate the feedback but Cordova support is very important to us so developers can use the full ecosystem of plugins across Cordova and Capacitor, and we plan to continue to support Cordova plugins moving forward.

**Reply by Samnan** [reply]
Thanks Mate
